src/server/pfs/server/http.go

```diff
@@ -60,8 +62,20 @@ func (s *HTTPServer) getFileHandler(w http.ResponseWriter, r *http.Request, ps h
 	}
 	filePaths := strings.Split(ps.ByName("filePath"), "/")
 	fileName := filePaths[len(filePaths)-1]
+	ctx := context.Background()
+	for _, cookie := range r.Cookies() {
+		if cookie.Name == auth.ContextTokenKey {
+			ctx = metadata.NewIncomingContext(
+				ctx,
+				metadata.New(
```

Comment:
very minor suggestion: you could also write this as `metadata.Pairs(auth.ContextTokenKey, cookie.Value)` and it might be a little shorter